optex/base/optex.lua

```diff
@@ -483,40 +485,69 @@ end
 -- We use the `node.direct` way of working with nodes. This is less safe, and
 -- certainly not idiomatic Lua, but faster and codewise more close to the way
 -- \TeX/ works with nodes.
---
+local function is_color_needed(head, n, id, subtype) -- returns non-stroke, stroke color needed
+    if id == glyph_id then
+        return true, true
+    end
+    if id == whatsit_id and
+            (subtype == pdfliteral_id
+            or subtype == pdfsave_id
+            or subtype == pdfrestore_id) then
+        return true, true
+    end
+    if id == glue_id then
+        n = getleader(n)
+        if not n then
+            -- ordinary "space" glue
+            return false, false
+        else
+            id = getid(n)
+            if id == hlist_id or id == vlist_id then
+                -- leaders with hlist/vlist get single color
+                return true, false
+            end
+            -- else we have a leader rule checked below (note that color will
```

Comment:
why there isn't return true, true ? We are before leaders with rule and we dont know if the rule will be reealized by stroke or fill operator.

Reply:
You are right. E.g. the following doesn't work as expected (may required adjustments for different fonts):

```
\hbox to 11pt{\Red a\Green \leaders\hrule width5pt height5pt depth5pt\hfil \Brown a}
```

`true, true` is the simplest here. But I still don't like the overall structue and the function itself. I am thinking about simplifying and inlining it into `colorize` as it was before.